interface/controller/pkmn_controller.go

```diff
@@ -5,26 +5,32 @@ import (
 	"strconv"
 
 	"github.com/jvillarreal-w/academy-go-q42021/domain/model"
+	"github.com/jvillarreal-w/academy-go-q42021/interface/context"
+	"github.com/jvillarreal-w/academy-go-q42021/interface/external"
 	"github.com/jvillarreal-w/academy-go-q42021/usecase/interactor"
 	u "github.com/jvillarreal-w/academy-go-q42021/utils"
 )
 
 type pokemonController struct {
 	pokemonInteractor interactor.PokemonInteractor
+	pokemonExternal   external.PokemonExternal
 }
 
 type PokemonController interface {
-	GetPokemon(c Context) error
-	GetPokemonById(c Context) error
+	GetPokemon(c context.Context) error
+	GetPokemonById(c context.Context) error
 }
 
-func NewPokemonController(pi interactor.PokemonInteractor) PokemonController {
-	return &pokemonController{pi}
+func NewPokemonController(pi interactor.PokemonInteractor, pe external.PokemonExternal) PokemonController {
+	return &pokemonController{pi, pe}
 }
 
-func (pc *pokemonController) GetPokemon(c Context) error {
+func (pc *pokemonController) GetPokemon(c context.Context) error {
 	var p []*model.Pokemon
 
+	external_pkmn, _ := external.NewPokemonExternal().GetExternalPokemon(p, c)
```

Comment:
You sould probably keep in a variable the result of `external.NewPokemonExternal()`

Reply:
If you're expecting an Error there, then remember to handle it